node/worker/worker.go

```diff
@@ -300,10 +306,56 @@ func (w *Worker) IncomingReceipts() []*types.CXReceiptsProof {
 	return w.current.incxs
 }
 
+// CollectAndVerifySlashes ..
+func (w *Worker) CollectAndVerifySlashes() error {
+	allSlashing, err := w.chain.ReadPendingSlashingCandidates()
```

Comment:
where do you remove the one that's proposed from pending?

Reply:
i remove them all in writeblockwithstate, 

```
			if err := bc.UpdateBlockRewardAccumulator(
				batch, payout, block.Number().Uint64(),
			); err != nil {
				return NonStatTy, err
			}
			if err := bc.DeletePendingSlashingCandidates(); err != nil {
				return NonStatTy, err
			}
```